src/execution_plan/execution_plan_build/execution_plan_util.c

```
@@ -123,74 +137,226 @@ void ExecutionPlan_LocateOps
+		OPType t = OpBase_Type(op);
+
+		// make sure we're allowed to inspect current op
+		if(op != root && (t == OPType_PROJECT || t == OPType_AGGREGATE)) {
+			array_append(taps, op);
+			continue;
+		}
+
+		// add operation's children to queue
+		uint n = OpBase_ChildCount(op);
+		if(n == 0) {
+			array_append(taps, op);
```

Cppcheck (2.10-2)

[error] Common realloc mistake

(memleakOnRealloc)